From 0f21259f721cf392bd2f7d496de4f85f8eebd19d Mon Sep 17 00:00:00 2001
From: Aaron Kling <webgeek1234@gmail.com>
Date: Fri, 25 Aug 2023 20:10:07 -0500
Subject: [PATCH] Convert some display libraries to blueprint

So other bp rules can reference it. Based on sm8350

Change-Id: Ia62eed4e790166f72324b41c2360a2727b2c056c
---
 libdrmutils/Android.bp    | 25 +++++++++++++++++++++++++
 libdrmutils/Android.mk    | 15 ---------------
 sdm/libs/utils/Android.bp | 20 ++++++++++++++++++++
 sdm/libs/utils/Android.mk | 17 -----------------
 4 files changed, 45 insertions(+), 32 deletions(-)
 create mode 100644 libdrmutils/Android.bp
 delete mode 100644 libdrmutils/Android.mk
 create mode 100644 sdm/libs/utils/Android.bp
 delete mode 100644 sdm/libs/utils/Android.mk

diff --git a/libdrmutils/Android.bp b/libdrmutils/Android.bp
new file mode 100644
index 00000000..5a557149
--- /dev/null
+++ b/libdrmutils/Android.bp
@@ -0,0 +1,25 @@
+cc_library_shared {
+
+    name: "libdrmutils",
+    vendor: true,
+    header_libs: [
+        "qti_kernel_headers",
+    ],
+    shared_libs: [
+        "libdrm",
+        "libdl",
+    ],
+    cflags: [
+        "-DLOG_TAG=\"DRMUTILS\"",
+        "-Wall",
+        "-Werror",
+        "-fno-operator-names",
+    ],
+
+    srcs: [
+        "drm_master.cpp",
+        "drm_res_mgr.cpp",
+        "drm_lib_loader.cpp",
+    ],
+
+}
diff --git a/libdrmutils/Android.mk b/libdrmutils/Android.mk
deleted file mode 100644
index 773fa634..00000000
--- a/libdrmutils/Android.mk
+++ /dev/null
@@ -1,15 +0,0 @@
-LOCAL_PATH := $(call my-dir)
-include $(CLEAR_VARS)
-
-LOCAL_MODULE                  := libdrmutils
-LOCAL_VENDOR_MODULE           := true
-LOCAL_MODULE_TAGS             := optional
-LOCAL_C_INCLUDES              := external/libdrm \
-                                 $(TARGET_OUT_INTERMEDIATES)/KERNEL_OBJ/usr/include
-LOCAL_SHARED_LIBRARIES        := libdrm libdl
-LOCAL_CFLAGS                  := -DLOG_TAG=\"DRMUTILS\" -Wall -std=c++11 -Werror -fno-operator-names
-LOCAL_CLANG                   := true
-LOCAL_ADDITIONAL_DEPENDENCIES := $(TARGET_OUT_INTERMEDIATES)/KERNEL_OBJ/usr
-LOCAL_SRC_FILES               := drm_master.cpp drm_res_mgr.cpp drm_lib_loader.cpp
-
-include $(BUILD_SHARED_LIBRARY)
diff --git a/sdm/libs/utils/Android.bp b/sdm/libs/utils/Android.bp
new file mode 100644
index 00000000..154658c8
--- /dev/null
+++ b/sdm/libs/utils/Android.bp
@@ -0,0 +1,20 @@
+
+cc_library_shared {
+
+    name: "libsdmutils",
+    defaults: ["display_defaults"],
+    vendor: true,
+
+    header_libs: ["display_headers"],
+    cflags: [
+        "-Wno-implicit-float-conversion",
+        "-DLOG_TAG=\"SDM\""
+    ],
+    srcs: [
+        "debug.cpp",
+        "rect.cpp",
+        "sys.cpp",
+        "formats.cpp",
+        "utils.cpp",
+    ],
+}
diff --git a/sdm/libs/utils/Android.mk b/sdm/libs/utils/Android.mk
deleted file mode 100644
index f2e5e8bc..00000000
--- a/sdm/libs/utils/Android.mk
+++ /dev/null
@@ -1,17 +0,0 @@
-LOCAL_PATH := $(call my-dir)
-include $(CLEAR_VARS)
-include $(LOCAL_PATH)/../../../common.mk
-
-LOCAL_MODULE                  := libsdmutils
-LOCAL_VENDOR_MODULE           := true
-LOCAL_MODULE_TAGS             := optional
-LOCAL_C_INCLUDES              := $(common_includes)
-LOCAL_HEADER_LIBRARIES        := display_headers
-LOCAL_CFLAGS                  := -DLOG_TAG=\"SDM\" $(common_flags)
-LOCAL_SRC_FILES               := debug.cpp \
-                                 rect.cpp \
-                                 sys.cpp \
-                                 formats.cpp \
-                                 utils.cpp
-
-include $(BUILD_SHARED_LIBRARY)
-- 
2.47.0

